controllers/v1/project/templates.js

```diff
@@ -0,0 +1,192 @@
+                    message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+                    errorObject: error
+                });
+            }
+        })
+    }
+
+    /**
+    * @api {post} /improvement-project/api/v1/project/templates/bulkUpdate 
+    * Bulk Update projects templates.
+    * @apiVersion 1.0.0
+    * @apiGroup Project Templates
+    * @apiParam {File} projectTemplates Mandatory project templates file of type CSV.
+    * @apiSampleRequest /improvement-project/api/v1/project/templates/bulkUpdate
+    * @apiUse successBody
+    * @apiUse errorBody
+    */
+
+      /**
+      * Bulk Update project templates
+      * @method
+      * @name bulkUpdate
+      * @returns {JSON} returns uploaded project templates.
+     */
+
+    async bulkUpdate(req) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                if ( !req.files || !req.files.projectTemplates ) {
+                    return resolve(
+                      {
+                        status : HTTP_STATUS_CODE["bad_request"].status, 
+                        message : CONSTANTS.apiResponses.PROJECT_TEMPLATES_CSV
+                      }
+                    )
+                }
+                
+                const templatesData = 
+                await csv().fromString(req.files.projectTemplates.data.toString());
+
+                const projectTemplates = await projectTemplatesHelper.bulkUpdate(
+                    templatesData,
+                    req.userDetails.userInformation.userId
+                );
+
+                return resolve(projectTemplates);
+
+            } catch (error) {
+                return reject({
+                    status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+                    message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+                    errorObject: error
+                });
+            }
+        })
+    }
+
+     /**
+    * @api {post} /improvement-project/api/v1/project/templates/importFromTemplates/:projectTemplateExternalId 
+    * Import templates from existsing project templates.
+    * @apiVersion 1.0.0
+    * @apiGroup Project Templates
+    * @apiSampleRequest /improvement-project/api/v1/project/templates/importFromTemplates/template-1
+    * @apiParamExample {json} Request: 
+    * {
+    * "externalId" : "template1"
+    * }
+    * @apiParamExample {json} Response:
+    * {
+    "message": "Successfully created duplicate project templates",
+    "status": 200,
+    "result": {
+        "_id": "5f2402f570d11462f8e9a591"
+    },
+    "success": true
+    }
+    * @apiUse successBody
+    * @apiUse errorBody
+    */
+
+      /**
+      * Import templates from existsing project templates.
+      * @method
+      * @name importFromTemplates
+      * @returns {JSON} returns imported project templates.
+     */
+
+    async importFromTemplates(req) {
```

Comment:
@aman-tunerlabs - This can be moved to userProjects when imported from the library by the user on the app, and here we can have API just for pure import from template and map template to solution etc. Let me know if you want to discuss this.